curator-framework/src/main/java/org/apache/curator/framework/imps/EnsembleTracker.java

```diff
@@ -170,8 +170,10 @@ public static String configToConnectionString(QuorumVerifier data) throws Except
             {
                 sb.append(",");
             }
-            InetSocketAddress address = Objects.firstNonNull(server.clientAddr, server.addr);
-            sb.append(address.getAddress().getHostAddress()).append(":").append(address.getPort());
+            InetSocketAddress clientAddress = server.clientAddr;
```

Comment:
You are right. CURATOR-392 looks like a more complete attempt at fixing this, so this PR doesn't seem necessary.